src/differentiable_functions.jl

```diff
@@ -113,3 +113,22 @@ function DifferentiableSparseMultivariateFunction(f!::Function, g!::Function)
     end
     return DifferentiableSparseMultivariateFunction(f!, g!, fg!)
 end
+
+
+immutable DifferentiableGivenSparseMultivariateFunction{Tv, Ti} <: AbstractDifferentiableMultivariateFunction
+    f!::Function
+    g!::Function
+    fg!::Function
+    J::SparseMatrixCSC{Tv, Ti}
+end
+
+alloc_jacobian(df::DifferentiableGivenSparseMultivariateFunction, args...) = deepcopy(df.J)
```

Comment:
Otherwise, `df.J` will be modified within the algorithm, right? Would not that be error-prone?